sdks/java/core/src/main/java/org/apache/beam/sdk/coders/DelegateCoder.java

```diff
@@ -101,8 +104,28 @@ public Object structuralValue(T value) throws Exception {
   }
 
   @Override
+  public boolean equals(Object o) {
+    if (o == null || this.getClass() != o.getClass()) {
```

Comment:
instanceof automatically removes the need for the null check
if (!(o instanceof DelegateCoder)) {
  return false;
}